### PR TITLE
PID class fixes

### PR DESCRIPTION
**Pull request by katiewang22**

### CODE REVIEW

**Comment by ChristianPawlowicz** [comment]
While this would address a lot of the issues we have on the board, you are covering too many topics at once. Pull requests should only address one thing at a time. Fixing one bug, adding one new feature, cleaning up one class, etc. This is not just done for the benefit of clear commit history, but also to allow reviewers like myself to provide good feedback on the changes being made. When you create a large pull request like this that changes a bunch of things at once, I have to look at every single change you made and guess as to the context of why you are changing it. I might need to reach out to you to get a clearer picture, which makes the process more difficult. Ideally, I would get all the context I need from the title of your PR. For example, if the title was just "PID class fixes" I would expect to only see changes to the PID class and any function calls to that class. But you have "and other code cleanup" in your title as well, which is just glossing over a whole bunch of other changes. Break down the changes you made here into separate PRs so that I can get good context on every change you are making.

**Comment by katiewang22** [comment]
I fixed this PR to only have the changes to the PID class.

File: src/main/java/frc/robot/lib/RoboLionsPID.java [hunk 1]
```diff
@@ -100,60 +97,39 @@ public double execute(double command, double feedback) {
...
-       deadband_active = false; //true
-       //integrate if error is small and we are close to our command
-    }
-    */
-
-    // 7. Calculate PID Control Equation
+    // 6. Calculate PID Control Equation
     output = error*proportionalGain + integral_charge*integralGain - derivativeCalculation*derivativeGain;
-    if(output > maxOutput) {
+    if (output > maxOutput) {
```
Comment (ChristianPawlowicz) [inline src/main/java/frc/robot/lib/RoboLionsPID.java]: Just like you did on line 110, you can simplify this if else case by utilizing Math.min / Math.max.

File: src/main/java/frc/robot/lib/RoboLionsPID.java [hunk 2]
```diff
@@ -100,60 +97,39 @@ public double execute(double command, double feedback) {
...
     if (enableDeadBand == true) {
-      if((error < upperDeadbandLimit) && (error > lowerDeadbandLimit)) {
-        deadband_counter++;
-        if(deadband_counter > 13) { // 13 = 50 / 4 = 50 ms split into 4 quarters
-          deadband_active = true;
-          deadband_counter = 13;
-        }
-          // integral_charge = integral_charge + error*deltaTime;
+      if ((error <= upperDeadbandLimit) && (error >= lowerDeadbandLimit)) {
+        return output;
```
Comment (ChristianPawlowicz) [inline src/main/java/frc/robot/lib/RoboLionsPID.java]: The return value should be 0.0, be explicit.

File: src/main/java/frc/robot/lib/RoboLionsPID.java [hunk 3]
```diff
@@ -4,46 +4,43 @@ public class RoboLionsPID {
...
   // These are used to turn off the limiter cage and the deadband 2/15/20
   public boolean enableCage = true;
   public boolean enableDeadBand = true;
 
-  public static int deadband_counter = 0;
-
   // Called just before this Command runs the first time to set your values
   public void initialize(double _P,double _I, double _D,
-                         double Cage_Limit,double Deadband,double MaxOutput) {
+                         double Cage_Limit, double Deadband, double MaxOutput) {
```
Comment (ChristianPawlowicz) [inline src/main/java/frc/robot/lib/RoboLionsPID.java]: Why even have a deadband and cage limit parameter in the constructor if you are not going to use it in the execution function?

Comment (katiewang22) [reply src/main/java/frc/robot/lib/RoboLionsPID.java]: Wait the parameters get used for the upper/lower limits

Comment (ChristianPawlowicz) [reply src/main/java/frc/robot/lib/RoboLionsPID.java]: Yes, you set the limits. But both the flags that you use to check those limits are set to false. Thus they are never used. You should remove the limit initializations here and put them in the other constructor.

File: src/main/java/frc/robot/lib/RoboLionsPID.java [hunk 4]
```diff
@@ -100,60 +97,36 @@ public double execute(double command, double feedback) {
     // 3. Latch internal error's state for next time
     previousError = error; 
 
-    // 4. If in deadband, don't integrate the error
+    // 4. If in deadband, PID output = 0
     if (enableDeadBand == true) {
```
Comment (ChristianPawlowicz) [inline src/main/java/frc/robot/lib/RoboLionsPID.java]: Make this just if(enableDeadBand). No need for the == true

File: src/main/java/frc/robot/lib/RoboLionsPID.java [hunk 5]
```diff
@@ -100,60 +97,36 @@ public double execute(double command, double feedback) {
...
-          // integral_charge = integral_charge + error*deltaTime;
+      if ((error <= upperDeadbandLimit) && (error >= lowerDeadbandLimit)) {
+        return 0.0;
       }
     }
     
-    // 5. If in Zone of Contrability, overcome any frictional forces
+    // 5. If in cage limit, prevent integral from increasing further
+    integral_charge += error*deltaTime;
     if (enableCage == true) {
```
Comment (ChristianPawlowicz) [inline src/main/java/frc/robot/lib/RoboLionsPID.java]: Same here. Remove == true

**Pull request merged**